src/index.js

```diff
@@ -76,13 +76,20 @@ const translateArrayIntoLink = (linkList, params) => {
     null;
 };
 
-export default (object, linkName, params) => {
+export default (source, linkName, params) => {
   if (!linkName) {
     throw new Error('No link name was passed to hal link resolver');
   }
-  if (object) {
+  if (source instanceof Array) {
+    const links = source.filter(item => item.rel === linkName);
+    if (links.length !== 0) {
+      return links[0].href ? links[0].href : null;
+    }
+    return null;
+  }
+  if (source) {
     // eslint-disable-next-line no-underscore-dangle
-    const links = object._links || object.links || object;
+    const links = source._links || source.links || source;
```

Comment:
I'd like the new functionality to still keep support of "autodiscovering" the links. This way, it is bypassed.

So if I have a user object
```javascript
const user = {
  links: [
    {
      rel: 'edit',
      link: '/user/1/edit',
    },
  ],
};
```

I should be able to do this:

```javascript
resolve(user, 'edit');
```

Without actually giving it the links object, but the whole user.